cmd/update/matchable_attribute_util.go

```diff
@@ -3,45 +3,196 @@ package update
 import (
 	"context"
 	"fmt"
+	"os"
 
 	sconfig "github.com/flyteorg/flytectl/cmd/config/subcommand"
+	cmdCore "github.com/flyteorg/flytectl/cmd/core"
+	cmdUtil "github.com/flyteorg/flytectl/pkg/commandutils"
 	"github.com/flyteorg/flytectl/pkg/ext"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
 )
 
-func DecorateAndUpdateMatchableAttr(ctx context.Context, project, domain, workflowName string,
-	updater ext.AdminUpdaterExtInterface, mcDecorator sconfig.MatchableAttributeDecorator, dryRun bool) error {
-	matchingAttr := mcDecorator.Decorate()
-	if len(workflowName) > 0 {
-		// Update the workflow attribute using the admin.
-		if dryRun {
-			fmt.Printf("skipping UpdateWorkflowAttributes request (dryRun)\n")
-		} else {
-			err := updater.UpdateWorkflowAttributes(ctx, project, domain, workflowName, matchingAttr)
-			if err != nil {
-				return err
-			}
-		}
-		fmt.Printf("Updated attributes from %v project and domain %v and workflow %v\n", project, domain, workflowName)
-	} else {
-		// Update the project domain attribute using the admin.
-		if dryRun {
-			fmt.Printf("skipping UpdateProjectDomainAttributes request (dryRun)\n")
-		} else {
-			if len(domain) == 0 {
-				err := updater.UpdateProjectAttributes(ctx, project, matchingAttr)
-				if err != nil {
-					return err
-				}
-				fmt.Printf("Updated attributes from %v project\n", project)
-			} else {
-				err := updater.UpdateProjectDomainAttributes(ctx, project, domain, matchingAttr)
-				if err != nil {
-					return err
-				}
-				fmt.Printf("Updated attributes from %v project and domain %v\n", project, domain)
-			}
-		}
+func DecorateAndUpdateMatchableAttr(
+	ctx context.Context,
+	cmdCtx cmdCore.CommandContext,
+	project, domain, workflow string,
+	resourceType admin.MatchableResource,
+	attributeDecorator sconfig.MatchableAttributeDecorator,
+	dryRun bool,
+	force bool,
+) error {
+	if project == "" {
+		return fmt.Errorf("project is required")
+	}
+	if domain == "" && workflow != "" {
+		return fmt.Errorf("domain is required")
+	}
+
+	switch {
+	case workflow != "":
+		return updateWorkflowMatchableAttributes(ctx, cmdCtx, project, domain, workflow, resourceType, attributeDecorator, dryRun, force)
+	case domain != "":
+		return updateProjectDomainMatchableAttributes(ctx, cmdCtx, project, domain, resourceType, attributeDecorator, dryRun, force)
+	default:
+		return updateProjectMatchableAttributes(ctx, cmdCtx, project, resourceType, attributeDecorator, dryRun, force)
+	}
+}
+
+func updateProjectMatchableAttributes(
+	ctx context.Context,
+	cmdCtx cmdCore.CommandContext,
+	project string,
+	resourceType admin.MatchableResource,
+	attributeDecorator sconfig.MatchableAttributeDecorator,
+	dryRun bool,
+	force bool,
+) error {
+	if project == "" {
+		panic("project is empty")
+	}
+
+	response, err := cmdCtx.AdminFetcherExt().FetchProjectAttributes(ctx, project, resourceType)
+	if err != nil && !ext.IsNotFoundError(err) {
```

Comment:
Looks like here you check `IsNotFoundError()` but in earlier updates you don't. I assume this is because there is no equivalent of create matchable attribute. I believe your test cases [like this one](https://github.com/flyteorg/flytectl/pull/423/files#diff-86a38b662eeac53bf732df664b4978552fa2a401198139beadc5a2ef551add67R248) cover this but just want to confirm.